src/Gaufrette/Adapter/GridFS.php

```diff
@@ -256,14 +256,21 @@ public function listKeys($prefix = '')
         return $result;
     }
 
+    /**
+     * {@inheritdoc}
+     */
     public function size($key)
     {
-        if (!$this->exists($key)) {
-            return false;
+        try {
+            $this->exists($key);
+        } catch (StorageFailure $e) {
+            throw new FileNotFound($key, $e->getCode(), $e);
```

Comment:
`exists` can return a `StorageFailure`. In such case, it should not be wrapped by a `FileNotFound` exception. IMHO you should keep the condition and throw an exception in its body.